packages/blade/src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,39 @@
+import type { ReactElement } from 'react';
+import type { IconComponent } from '..';
+
+import StyledIconButton from './StyledIconButton.web';
+import { makeAccessible } from '~src/utils/makeAccessible';
+
+export type IconButtonProps = {
+  /** Icon component to be rendered, eg. `CloseIcon` */
+  icon: IconComponent;
+  onClick: () => void;
+  size?: 'medium' | 'large'; // todo: discrepancy in size between code and figma
+  contrast?: 'low' | 'high';
+  /** **a11y**: Sets aria-label to help users know what the action does, eg 'Dismiss alert' */
+  accessibilityLabel: string;
+};
+
+/**
+ * Internal component for making clickable icons
+ */
+const IconButton = ({
+  icon: Icon,
+  onClick,
+  size = 'medium',
+  contrast = 'low',
```

Comment:
I was of the impression we were only scoping a transparent icon button as part of this, why include a `contrast` then? Is the usage of the `contrast` specific to Alerts?

Reply:
In our use cases in the dark background I see the `IconButton` will have a transparent background (like in Alerts). Maybe I am missing something?

<img width="164" alt="Screenshot 2022-08-05 at 5 57 37 PM" src="https://user-images.githubusercontent.com/2913308/183077525-c1af8c2c-1d7e-40e9-8c70-501c8603da27.png">

Reply:
For high contrast components, the colors are different:

| Low contrast | High contrast |
| - | - |
| `Action/Surface/Icon/lowContrast/Default`  | `Action/Surface/Icon/highContrast/Default` |
| <img width="613" alt="image" src="https://user-images.githubusercontent.com/6682655/183842850-8b0d34eb-f711-4d86-a5d7-e964e22c4f00.png"> | <img width="609" alt="image" src="https://user-images.githubusercontent.com/6682655/183842898-3ff7d2f1-1c63-4ec6-abed-d8a64d29ed45.png"> |

Reply:
With this approach, the consumer will also need to pass hover, active, focus colors, etc. Was replicating the [design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=12702%3A149930) here. In figma the tokens are coming from `action/surface` 🤔 /cc @saurav12

Reply:
After discussions, plan is to go ahead with current implementation and pick the colors off the new surface.action tokens. Will create a followup once the tokens are updated on figma and code.